pandas/io/excel/_openpyxl.py

```diff
@@ -409,7 +421,11 @@ def write_cells(
                 row=freeze_panes[0] + 1, column=freeze_panes[1] + 1
             )
 
+        n_cols = 0
+        n_rows = 0
         for cell in cells:
+            n_cols = max(n_cols, cell.col)
```

Comment:
Can these be inferred outside of the loop from the dimensions of the frame?

Reply:
There is quite some code translating a frame to excel cells, dealing with multiindexes etc. So this is not too straight forward. But that code in intertwined with the formatting code, so I considered the following options:

 - get size from frame, try to deal with edge cases for multiindex, index True/False, header True/False etc
 - get result from the get_formatted_cells iterator, run through it a second time
 - bypass the normal writer function altogether and use a separate, dedicated write_table function
 - extract the size from the writer function

I picked the latter

Reply:
Hmm OK. At least outside the loop though wouldn't the nrows be len(cells) and the ncols just be the length of any item within cells?

Reply:
unfortunately the cells are a 1D iterator of items (each with a row and col property), not a list of rows.